crates/networking/discv5/src/subnet.rs

```diff
@@ -1,173 +1,414 @@
 use alloy_rlp::{BufMut, Decodable, Encodable, bytes::Bytes};
 use anyhow::anyhow;
 use discv5::Enr;
-use ssz::{Decode, Encode};
-use ssz_derive::{Decode, Encode};
-use ssz_types::{BitVector, typenum::U64};
-use tracing::error;
+use ssz::Encode;
+use ssz_types::{
+    BitVector,
+    typenum::{U4, U64},
+};
+use tracing::{error, trace};
 
 pub const ATTESTATION_BITFIELD_ENR_KEY: &str = "attnets";
+pub const SYNC_COMMITTEE_BITFIELD_ENR_KEY: &str = "syncnets";
+pub const SYNC_COMMITTEE_SUBNET_COUNT: usize = 4;
 
-#[derive(Clone, Debug, PartialEq)]
-pub enum Subnet {
-    Attestation(u8),
-    SyncCommittee(u8),
+/// Represents the attestation subnets a node is subscribed to
+///
+/// This directly wraps a BitVector<U64> for the attestation subnet bitfield
+/// and handles encoding/decoding to raw bytes for ENR records.
+#[derive(Clone, Debug, Default)]
+pub struct AttestationSubnets(pub BitVector<U64>);
+
+impl AttestationSubnets {
+    /// Create a new empty attestation subnets bitfield
+    pub fn new() -> Self {
+        Self(BitVector::new())
+    }
+
+    /// Set a specific attestation subnet bit
+    pub fn set(&mut self, index: usize, value: bool) -> anyhow::Result<()> {
+        self.0
+            .set(index, value)
+            .map_err(|err| anyhow!("Subnet ID out of bounds: {err:?}"))
+    }
+
+    /// Get a specific attestation subnet bit
+    pub fn get(&self, index: usize) -> anyhow::Result<bool> {
+        self.0
+            .get(index)
+            .map_err(|err| anyhow!("Subnet ID out of bounds: {err:?}"))
+    }
+
+    pub fn enable_attestation_subnet(&mut self, subnet_id: u8) -> anyhow::Result<()> {
+        if subnet_id < 64 {
+            self.set(subnet_id as usize, true)?;
+        }
+        Ok(())
+    }
+
+    pub fn disable_attestation_subnet(&mut self, subnet_id: u8) -> anyhow::Result<()> {
+        if subnet_id < 64 {
+            self.set(subnet_id as usize, false)?;
+        }
+        Ok(())
+    }
```

Comment:
I think we shouldn't have both `set`  and `enable_attestation_subnet`/`disable_attestation_subnet`

ideally we have only one or the other